gammapy/background/make.py

```diff
@@ -0,0 +1,470 @@
+        ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
+                                                     [events_cube.energy_bins,
+                                                      events_cube.dety_bins,
+                                                      events_cube.detx_bins])
+        ev_cube_hist = Quantity(ev_cube_hist, '') # counts
+
+        # fill cube
+        events_cube.background += ev_cube_hist
+
+        # fill livetime for bins where E_max > E_thres
+        energy_max = events_cube.energy_bins[1:]
+        dummy_dety_max = np.zeros_like(events_cube.dety_bins[1:])
+        dummy_detx_max = np.zeros_like(events_cube.detx_bins[1:])
+        # define grid of max values (i.e. bin max values for each 3D bin)
+        energy_max, dummy_dety_max, dummy_detx_max = np.meshgrid(energy_max,
+                                                                 dummy_dety_max,
+                                                                 dummy_detx_max,
+                                                                 indexing='ij')
+        mask = energy_max > energy_threshold
+
+        # fill cube
+        livetime_cube.background += livetime*mask
+
+    return events_cube, livetime_cube
+
+
+def divide_bin_volume(cube):
+    """Divide by the bin volume.
+
+    Parameters
+    ----------
+    cube : `~gammapy.background.CubeBackgroundModel`
+        Cube containing the data to process.
+
+    Returns
+    -------
+    cube : `~gammapy.background.CubeBackgroundModel`
+        Cube divided by the bin volume.
+    """
+    delta_energy = cube.energy_bins[1:] - cube.energy_bins[:-1]
+    delta_y = cube.dety_bins[1:] - cube.dety_bins[:-1]
+    delta_x = cube.detx_bins[1:] - cube.detx_bins[:-1]
+    # define grid of deltas (i.e. bin widths for each 3D bin)
+    delta_energy, delta_y, delta_x = np.meshgrid(delta_energy, delta_y,
+                                                 delta_x, indexing='ij')
+    bin_volume = delta_energy.to('MeV')*(delta_y*delta_x).to('sr') # TODO: use TeV!!!
+    cube.background /= bin_volume
+
+    return cube
+
+
+def set_zero_level(cube):
+    """Setting level 0 to something very small.
+
+    Also for NaN values: they are in the 1st few E bins,
+    where no stat is present: (0 events/ 0 livetime = NaN)
+
+    Parameters
+    ----------
+    cube : `~gammapy.background.CubeBackgroundModel`
+        Cube containing the data to process.
+
+    Returns
+    -------
+    cube : `~gammapy.background.CubeBackgroundModel`
+        Cube with 0-level applied.
+    """
+    zero_level = Quantity(1.e-10, cube.background.unit)
+    zero_level_mask = cube.background < zero_level
+    cube.background[zero_level_mask] = zero_level
+    nan_mask = np.isnan(cube.background)
+    cube.background[nan_mask] = zero_level
+
+    return cube
+
+
+def smooth(bg_cube, n_counts):
+    """
+    Smooth background cube model.
+
+    Smooth method:
+
+    1. slice model in energy bins: 1 image per energy bin
```

Comment:
Sorry, no time for this right now. Since we anyway are not sure this is the best smoothing method, I'm not sure it is worth the effort. Maybe in a later PR.